### PR TITLE
fix Issue 20126 - codegen reloads parameter from register when iasm c…

### PR DESCRIPTION
**Pull request by WalterBright**
…hanged the backing memory

### CODE REVIEW

**Comment by WalterBright** [comment]
Targeted master because Manu needs this and that's what he uses.

File: src/dmd/backend/cod1.d [hunk 1]
```diff
@@ -74,6 +74,7 @@ targ_size_t paramsize(elem *e, tym_t tyf);
  */
 bool regParamInPreg(Symbol* s)
 {
+    //printf("regPAramInPreg %s\n", s.Sident.ptr);
```
Comment (ErnyTech) [inline src/dmd/backend/cod1.d]: I think it should be removed

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/dmd/backend/cod1.d (hunk 1 above)
Comment (wilzbach) [reply src/dmd/backend/cod1.d]: No these debug outputs will come in handy in the future. It's a common practice to do this for DMD.

**Comment by TurkeyMan** [comment]
...LGTM?

**Comment by TurkeyMan** [comment]
`ip-172-31-7-111` strikes again!

**Comment by WalterBright** [comment]
How about someone approving this so Manu can move forward?

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**